xfields/beam_elements/beambeam3d.py

```diff
@@ -46,6 +64,7 @@ class BeamBeamBiGaussian3D(xt.BeamElement):
         'num_slices_other_beam': xo.Int64,
 
         'slices_other_beam_num_particles': xo.Float64[:],
+        'slices_other_beam_num_macroparticles': xo.Float64[:],
```

Comment:
If the slice contains 0 particles, the mean and sigma are 0. If the slice contains 1 particle, the sigma is 0. If the slice contains 2 particles, the uncoupling of the sigmas returns a nan because in the uncoupled frame the one of the sigmas will be 0.

Reply:
As discussed with @xbuffat, this Gould be handled when computing the momenta

Reply:
Introduced an if after `Sigmas_propagate()`:
<pre>
// apply kick only if Sig_11_hat_star != 0 and Sig_33_hat_star !=0 (corresponds to num_macroparts_in_slice > 2)
if (Sig_11_hat_star == 0 || Sig_33_hat_star ==0){
break;    
} 
</pre>